pkg/controller/core/workload_controller.go

```diff
@@ -170,6 +174,7 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 
 	if len(wl.OwnerReferences) == 0 && !wl.DeletionTimestamp.IsZero() {
 		// manual deletion triggered by the user
+		// Continue with normal finalizer removal
```

Comment:
This doesn't seem related to any of these changes.

I would probably revert or do this as a separate PR if you found it really necessary.